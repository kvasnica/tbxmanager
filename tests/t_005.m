function t_005
% tbxmanager enable/disable/restorepath

onCleanup(@(x) tbx_setupTest('done')); tbx_setupTest('start');

% set up the test repo
tbxmanager source add http://control.ee.ethz.ch/~mpt/tbx/test/test.xml
tbxmanager so re http://www.tbxmanager.com/package/index.xml
tbxmanager show sources

% nothing should be installed by default
tbxmanager show installed

% nothing should be enabled by default
tbxmanager show enabled
tbxmanager sh en

% install some packages
tbxmanager install tbx1 tbx2 tbx3

% they should be enabled
tbxmanager show enabled
% double-check by displaying tbxenabled.txt
type tbxenabled.txt
% double-check the path
assert(~isempty(strfind(which('tbx1_v1.m'), 'tbxstorage_test')));
assert(~isempty(strfind(which('tbx2_v1.m'), 'tbxstorage_test')));
assert(~isempty(strfind(which('tbx3.m'), 'tbxstorage_test')));

% disable one
tbxmanager disable tbx3
tbxmanager show enab
% double-check
type tbxenabled.txt
assert(isempty(strfind(which('tbx3.m'), 'tbxstorage_test')));

% disable for the second time
tbxmanager dis tbx3
tbxmanager sh en

% enable back
tbxmanager en tbx3
tbxmanager sh en
type tbxenabled.txt
assert(~isempty(strfind(which('tbx3.m'), 'tbxstorage_test')));

% enable the same
tbxmanager en tbx3
assert(~isempty(strfind(which('tbx3.m'), 'tbxstorage_test')));

% disable all
tbxmanager dis tbx2 tbx3
tbxmanager sh en
assert(~isempty(strfind(which('tbx1_v1.m'), 'tbxstorage_test')));
assert(isempty(strfind(which('tbx2_v1.m'), 'tbxstorage_test')));
assert(isempty(strfind(which('tbx3.m'), 'tbxstorage_test')));
type tbxenabled.txt

% enable all
tbxmanager enable tbx1 tbx2 tbx3

% manually remove from the path
w=warning; warning('off');
rmpath(genpath([fileparts(which(mfilename)) '/tbxstorage_test/']));
warning(w);
assert(isempty(strfind(which('tbx1_v1.m'), 'tbxstorage_test')));
assert(isempty(strfind(which('tbx2_v1.m'), 'tbxstorage_test')));
assert(isempty(strfind(which('tbx3.m'), 'tbxstorage_test')));

% restore path
tbxmanager restorepath
% all should be back
assert(~isempty(strfind(which('tbx1_v1.m'), 'tbxstorage_test')));
assert(~isempty(strfind(which('tbx2_v1.m'), 'tbxstorage_test')));
assert(~isempty(strfind(which('tbx3.m'), 'tbxstorage_test')));

end
