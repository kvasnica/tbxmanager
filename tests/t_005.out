--- tbxmanager source add http://control.ee.ethz.ch/~mpt/tbx/test/test.xml 
--- tbxmanager so re http://www.tbxmanager.com/package/index.xml 
--- tbxmanager show sources 
Active sources:

http://control.ee.ethz.ch/~mpt/tbx/test/test.xml
--- tbxmanager show installed 
Locally installed packages:

--- tbxmanager show enabled 
List of enabled packages:

--- tbxmanager sh en 
List of enabled packages:

--- tbxmanager install tbx1 tbx2 tbx3 

Installing version "1.0" of "tbx1"...
Downloading "http://control.ee.ethz.ch/~mpt/tbx/test/tbx1_v1.zip"...
"tbx1:1.0:all" installed to /Users/michal/matlab/mpt3/hg/tbxmgr-mk/tests/tbxstorage_test/tbx1/1.0/all
Toolbox "tbx1:1.0:all" added to the Matlab path.

Installing version "1.0" of "tbx2"...
Downloading "http://control.ee.ethz.ch/~mpt/tbx/test/tbx2_v1.zip"...
"tbx2:1.0:all" installed to /Users/michal/matlab/mpt3/hg/tbxmgr-mk/tests/tbxstorage_test/tbx2/1.0/all
Toolbox "tbx2:1.0:all" added to the Matlab path.

Installing version "2.0" of "tbx3"...
Downloading "http://control.ee.ethz.ch/~mpt/tbx/test/tbx3_v2.tgz"...
"tbx3:2.0:all" installed to /Users/michal/matlab/mpt3/hg/tbxmgr-mk/tests/tbxstorage_test/tbx3/2.0/all
Toolbox "tbx3:2.0:all" added to the Matlab path.
--- tbxmanager show enabled 
List of enabled packages:

tbx1   Version 1.0
tbx2   Version 1.0
tbx3   Version 2.0

tbx1:1.0:all
tbx2:1.0:all
tbx3:2.0:all

--- tbxmanager disable tbx3 
Toolbox "tbx3:2.0:all" removed from the Matlab path.
--- tbxmanager show enab 
List of enabled packages:

tbx1   Version 1.0
tbx2   Version 1.0

tbx1:1.0:all
tbx2:1.0:all

--- tbxmanager dis tbx3 
Toolbox "tbx3:2.0:all" removed from the Matlab path.
--- tbxmanager sh en 
List of enabled packages:

tbx1   Version 1.0
tbx2   Version 1.0
--- tbxmanager en tbx3 
Toolbox "tbx3:2.0:all" added to the Matlab path.
--- tbxmanager sh en 
List of enabled packages:

tbx1   Version 1.0
tbx2   Version 1.0
tbx3   Version 2.0

tbx1:1.0:all
tbx2:1.0:all
tbx3:2.0:all

--- tbxmanager en tbx3 
Toolbox "tbx3:2.0:all" added to the Matlab path.
--- tbxmanager dis tbx2 tbx3 
Toolbox "tbx2:1.0:all" removed from the Matlab path.
Toolbox "tbx3:2.0:all" removed from the Matlab path.
--- tbxmanager sh en 
List of enabled packages:

tbx1   Version 1.0

tbx1:1.0:all

--- tbxmanager enable tbx1 tbx2 tbx3 
Toolbox "tbx1:1.0:all" added to the Matlab path.
Toolbox "tbx2:1.0:all" added to the Matlab path.
Toolbox "tbx3:2.0:all" added to the Matlab path.
--- tbxmanager restorepath 
Toolbox "tbx1:1.0:all" added to the Matlab path.
Toolbox "tbx2:1.0:all" added to the Matlab path.
Toolbox "tbx3:2.0:all" added to the Matlab path.
